lock-api/src/main/java/com/palantir/lock/client/timestampleases/TimestampLeaseAcquirerImpl.java

```diff
@@ -158,22 +174,28 @@ private boolean wasRequestFullyFulfilled(RequestAndResponse requestAndResponse)
                     SafeArg.of("response", requestAndResponse.response()));
         }
 
-        boolean wasFullyFulfilled = requests.keySet().stream().allMatch(timestampName -> {
-            int requestedTimestamps = requests.get(timestampName);
-            long returnedTimestamps =
-                    responses.get(timestampName).getFreshTimestamps().getCount();
-            return returnedTimestamps >= requestedTimestamps;
-        });
+        long wereNotFullyFulfilled = requests.keySet().stream()
+                .filter(timestampName -> {
+                    int requestedTimestamps = requests.get(timestampName);
+                    long returnedTimestamps = responses.get(timestampName).getFreshTimestamps().getCount();
+                    if (returnedTimestamps >= requestedTimestamps) {
+                        return false;
+                    }
+                    telemetryTimestampShortFallConsumer.accept(requestedTimestamps - returnedTimestamps);
```

Comment:
I only want to report shortfall > 0

An alternative is to report shortfall >= 0, or any possible shortfall (or overflow)

Reply:
Realistically, we care about the count here rather than the histogram. But, it may be needed eventually

Reply:
I think we will struggle with interpreting these. If you care about magnitude, just log it. But I'd just log a binary based on ```wereNotFullyFulfilled > 0```

Reply:
and if that turns out to be false and we need more, we'll do that later, but at this point I'm struggling to think why I'd want to see the exact number.

Reply:
> Why do we need a histogram?
More data? Most appropriate would be a meter then? I don't usually like counters because of how they are reported but can use that as well

Reply:
yeah, just a counter